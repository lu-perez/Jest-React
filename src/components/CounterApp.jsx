import { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ initValue }) => {
  const [value, setValue] = useState(initValue)

  const handleAdd = () => { setValue(prev => prev + 1) }
  const handleSubstract = () => { setValue(prev => prev - 1) }
  const reset = () => { setValue(initValue) }

  return (
    <>
      <h1>CounterApp</h1>
      <h2> { value } </h2>
      <button onClick={handleAdd}> +1 </button>
      <button onClick={handleSubstract}> -1 </button>
      <button aria-label="reset" onClick={reset}> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
  initValue: PropTypes.number.isRequired
}

export default CounterApp
