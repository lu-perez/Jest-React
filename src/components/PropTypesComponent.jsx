import PropTypes from 'prop-types'

export const PropTypesComponent = ({ title, subtitle, someData }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>{someData}</p>
    </>
  )
}

PropTypesComponent.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number
}

PropTypesComponent.defaultProps = {
  // title: 'No title',
  someData: 'data'
}
