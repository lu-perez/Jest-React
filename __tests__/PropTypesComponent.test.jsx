import { render, screen } from '@testing-library/react'
import { PropTypesComponent } from '../src/components/PropTypesComponent'

describe('Pruebas en <PropTypesComponent />', () => {
  const title = 'Hola, soy Goku'

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<PropTypesComponent title={title}/>)
    expect(container).toMatchSnapshot()
  })

  test('debe mostrar el mensaje "Hola, soy Goku"', () => {
    render(<PropTypesComponent title={title}/>)
    expect(screen.getByText(title)).toBeTruthy()
    // screen.debug()
  })

  test('debe mostrar el tÃ­tulo en un h1', () => {
    render(<PropTypesComponent title={title}/>)
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
  })
})
