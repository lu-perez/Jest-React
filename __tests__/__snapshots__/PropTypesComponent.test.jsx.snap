// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PropTypesComponent /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p />
  <p>
    data
  </p>
</div>
`;
